Core/ResponseHandler.php

```diff
@@ -100,6 +100,11 @@ private static function _saveTransaction($oResponse, $oOrder, $oPayment)
         $oTransaction->wdoxidee_ordertransactions__date = new Field(
             Helper::getFormattedDbDate($oResponse->getData()['completion-time-stamp'])
         );
+        $oTransaction->wdoxidee_ordertransactions__validsignature = new Field($oResponse->isValidSignature());
+
+        if (!$oResponse->isValidSignature()) {
+            $this->_oLogger->warning('Transaction was possibly manipulated');
```

Comment:
_oLogger is not decelerated globally to be accessible outside of the constructor. Also the logging in the save method makes no sense. Here only the transaction is saved. Log it where you check for success, failure...